### PR TITLE
Add project picker empty state

### LINKED ISSUE #5567: project picker empty state
when you click the project picker dropdown and don't have access to any projects, we should render helpful information about joining a team or creating a project

### PR DESCRIPTION
**Pull request by billyvg**
Closes #5567 

* Removes filter if there are no projects
* Needs styling for "Create project" link and the parent menu item
* Do we want actionable items for creating/joining teams?

Edit: Added screenshots

No projects state
image

No projects when using filter
image

nochanges

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">You need to update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by benvinegar** [comment]
@ckj – any recommendations on what to do here style-wise?

**Comment by ckj** [comment]
@billyvg I'd probably have the no projects message, a divider, then a primary 'Create project' button with `.btn-block` treatment.

Nice PR!

File: src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx [hunk 1]
```diff
@@ -294,22 +335,31 @@ const ProjectSelector = React.createClass({
...
+                  onKeyUp={this.onKeyUp}
+                  onKeyDown={this.onKeyDown}
+                  onBlur={this.onFilterBlur}
+                  ref="filter"
+                />
+              </li>}
+
+            {this.renderProjectList({
+              organization: org,
+              hasProjectWrite: access.has('project:write'),
```
Comment (macqueen) [inline src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx]: i think this is unused now

**Review by macqueen — APPROVED** [review]
js looks good! i'll defer to chris on the styling

File: src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx (hunk 1 above)
Comment (billyvg) [inline src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx]: Oops, good catch, this *should* be used, was incorrectly using 'team:write'

**Comment by billyvg** [comment]
Added "Create team" and "Join team" depending on if you have `project:write` access and if there are existing teams, is this too much?
image

**Comment by ckj** [comment]
@billyvg I think we can get away with only the 'Create Project' button here. We're actually actively working on getting rid of the organization dashboard and redirecting a user to their last viewed project, as well as doing more to push people to setup a project earlier on. There will be edge cases, but the 'No projects' state will be way less of an issue very soon.

**Review by ckj — APPROVED** [review]
♠️

**Comment by ehfeng** [comment]
Love this.

**Pull request merged**